Bluetooth: GATT: Add support for Read Using Characteristic UUID

tedd-an opened the pull request:
This patch adds support for Read Using Characteristic UUID which is one
of the procedure to read the characteristic value especially when the
client only knows the characteristic UUID and does not know the handle
of the characteristic.

Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>

Vudentz commented on include/bluetooth/gatt.h:
```diff
@@ -985,6 +990,11 @@ struct bt_gatt_read_params {
+			u16_t start_handle;
+			u16_t end_handle;
+			struct bt_uuid *uuid;
+		} _uuid;
```
Id rename this from _uuid to by_uuid (iirc name shall not start with _).

Vudentz reviewed (commented):
Overall the patch looks good but Id split it in 3 parts: gatt, shell and tester.

tedd-an committed:
Bluetooth: GATT: Add support for Read Using Characteristic UUID

This patch adds support for Read Using Characteristic UUID which is one
of the procedure to read the characteristic value especially when the
client only knows the characteristic UUID and does not know the handle
of the characteristic.

Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>

tedd-an committed:
Bluetooth: shell: Add command for Read Using Characteristic UUID

This patch adds a command for Read Using Characteristic UUID.

Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>

tedd-an committed:
tests: bluetooth/tester: Add support Read Using Characteristic UUID

This patch adds an API to support Read Using Characteristic UUID.

Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>

MariuszSkamra reviewed (approved):
lgtm

Vudentz reviewed (approved):
LGTM
